.github/workflows/codeql_analysis.yml

```diff
@@ -1,30 +1,30 @@
-# For most projects, this workflow file will not need changing; you simply need
-# to commit it to your repository.
-#
-# You may wish to alter this file to override the set of languages analyzed,
-# or to provide custom queries or build logic.
 name: 'CodeQL'
 
 on:
   push:
-    branches:
-      - master
-      - next
+    branches: [ "master", "next", "master-bm" ]
   pull_request:
-    # The branches below must be a subset of the branches above
-    branches:
-      - master
-      - next
+    branches: [ "**" ]
 
 jobs:
   analyze:
-    name: Analyze
+    name: Analyze (${{ matrix.language }})
     runs-on: ubuntu-latest
 
+    permissions:
```

Comment:
Just comparing this to the monorepo, I think we need a couple more:

```
  pull-requests: write
  statuses: write
```
I think it requires these two as it writes comments on the PR itself if something is not right and also adds the status check.